subsystem/src/main/java/org/wildfly/extension/elytron/tls/subsystem/_private/WildFlyAcmeClient.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2018 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.extension.elytron.tls.subsystem._private;
+
+import static org.wildfly.extension.elytron.tls.subsystem._private.ElytronTLSLogger.LOGGER;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+
+import org.wildfly.common.Assert;
+import org.wildfly.security.manager.WildFlySecurityManager;
+import org.wildfly.security.x500.cert.acme.AcmeAccount;
+import org.wildfly.security.x500.cert.acme.AcmeChallenge;
+import org.wildfly.security.x500.cert.acme.AcmeClientSpi;
+import org.wildfly.security.x500.cert.acme.AcmeException;
+
+/**
+ * Implementation of the <a href="https://www.ietf.org/id/draft-ietf-acme-acme-12.txt">Automatic Certificate Management
+ * Environment (ACME)</a> protocol.
+ *
+ * @author <a href="mailto:fjuma@redhat.com">Farah Juma</a>
+ */
+public final class WildFlyAcmeClient extends AcmeClientSpi {
```

Comment:
This is a good example of a class that should be common code between the two subsystems. We can think about where to put this type of code.